fluent-syntax/src/parser.js

```diff
@@ -415,80 +415,83 @@ export default class FluentParser {
   }
 
   getPattern(ps) {
-    let buffer = '';
     const elements = [];
-    let firstLine = true;
+    ps.skipInlineWS();
 
-    if (this.withSpans) {
-      var spanStart = ps.getIndex();
+    // Special-case: trim leading whitespace and newlines.
+    if (ps.isPeekNextLinePattern()) {
+      ps.skipBlankLines();
+      ps.skipInlineWS();
```

Comment:
Yeah, I kept this to stay in line with other `isPeek...` methods which always reset the peek. We could try changing them such that they `skipToPeek` before returning `true`, although that would still need a bit of hand-holding in `isPeekNextLineVariantStart`.